instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import './CommentSection.css';
+
+class CommentSection extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            likes : 0,
+            liked : false,
+            newComment : { username: localStorage.getItem('username'), text: ""},
+            updatedComments : this.props.comments
+        }
+    }
+
+    componentDidMount() {
+        this.setState(prevState => { return {
+            likes: this.props.likes, 
+            updatedComments : JSON.parse(localStorage.getItem(`comments${this.props.id}`)),
+            newComment : { username: localStorage.getItem('username'), text: ""}
+        }});
+    }
+
+    inputChange = e => {
+        this.setState({ newComment : {username: localStorage.getItem('username'), text: e.target.value}});
+    }
+
+    commentCreation = e => {
+        e.preventDefault();
+        if(this.state.newComment.text === "") { return 'No text to post' }
+        let a = JSON.parse(localStorage.getItem(`comments${this.props.id}`));
+        a.push(this.state.newComment);
+        localStorage.setItem(`comments${this.props.id}`, JSON.stringify(a));
+        this.setState(prevState => { return { 
+            updatedComments: [...prevState.updatedComments, a],
```

Comment:
A few notes: 
- Nice job persisting comments
- Currently though, I need to refresh the page to see the updated comments
- This suggested change updates the comments w/o refresh.
```suggestion
            updatedComments: [...prevState.updatedComments, this.state.newComment],
```